compose/ui/ui/src/desktopMain/kotlin/androidx/compose/ui/awt/ComposeDialog.desktop.kt

```diff
@@ -45,91 +46,73 @@ import org.jetbrains.skiko.SkiaLayerAnalytics
  * ComposeDialog inherits javax.swing.JDialog.
  */
 class ComposeDialog : JDialog {
-    private val skiaLayerAnalytics: SkiaLayerAnalytics
     private val composePanel: ComposeWindowPanel
 
-    internal var rootForTestListener
-        get() = composePanel.rootForTestListener
-        set(value) { composePanel.rootForTestListener = value }
-
-    private fun createComposePanel() = ComposeWindowPanel(
+    private fun createComposePanel(
+        skiaLayerAnalytics: SkiaLayerAnalytics = SkiaLayerAnalytics.Empty,
+        savedState: SavedState? = null,
+    ) = ComposeWindowPanel(
         window = this,
         isUndecorated = ::isUndecorated,
         skiaLayerAnalytics = skiaLayerAnalytics,
+        savedState = savedState,
     )
 
-    constructor(
-        owner: Window?,
-        modalityType: ModalityType = ModalityType.MODELESS,
-        graphicsConfiguration: GraphicsConfiguration? = null
-    ) : super(owner, "", modalityType, graphicsConfiguration) {
-        skiaLayerAnalytics = SkiaLayerAnalytics.Empty
-        composePanel = createComposePanel()
-        contentPane.add(composePanel)
-    }
-
     /**
      * ComposeDialog is a dialog for building UI using Compose for Desktop.
      * ComposeDialog inherits javax.swing.JDialog.
      *
      * @param skiaLayerAnalytics Analytics that helps to know more about SkiaLayer behaviour.
      * SkiaLayer is underlying class used internally to draw Compose content.
      * Implementation usually uses third-party solution to send info to some centralized analytics gatherer.
+     * @param savedState The saved state to restore the UI state from a previous instance.
      */
     @ExperimentalComposeUiApi
     constructor(
-        owner: Window?,
+        owner: Window? = null,
```

Comment:
```suggestion
        owner: Window?,
```

We shouldn't make a default value for the `owner` paramter. JDialog has an issue - when we pass `owner = null` - we have different behavior, comparing to not passing `owner` at all (using another constructor). So, if we want to add `savedState`, we have to add into 2 constructors

Reply:
Besides removing it, we need to add `savedState` to the constructor without `owner`:
```
class ComposeDialog : JDialog {

@ExperimentalComposeUiApi
constructor(
  savedState: SavedState? = null,
  // fixing the previous mistake, we had to create this constructor, just for it
  skiaLayerAnalytics: SkiaLayerAnalytics = SkiaLayerAnalytics.Empty
): super() {
  
}

```

Reply:
Yep, already looking. Probably test will fail on this commit

Reply:
There is a wrong added constructor as well:
```
constructor(
        graphicsConfiguration: GraphicsConfiguration? = null,
    ) : this(
```
That don't accept `owner`, but reuses `super(owner = null). Better to remove it in general, but not in this PR